Solutions/Digital Shadows/Data Connectors/Digital Shadows/DigitalShadowsConnectorAzureFunction/DS_poller.py

```diff
@@ -0,0 +1,155 @@
+        json_obj = response.json()
+        for it in item:
+            for i in range(len(json_obj)):
+                if(it['source']['alert-id'] == json_obj[i]['id'] or it['source']['incident-id'] == json_obj[i]['id']):
+
+                    comment_data = self.DS_obj.get_triage_comments(it['id'])
+                    json_obj[i]['status'] = it['state']
+                    json_obj[i]['triage_id'] = it['id']
+                    json_obj[i]['triage_raised_time'] = it['raised']
+                    json_obj[i]['triage_updated_time'] = it['updated']
+                    
+                    comment_data_filtered = []
+                    for comment in comment_data:
+                        comment['user-name'] = comment['user']['name']
+                        del comment['triage-item-id']
+                        del comment['updated']
+                        del comment['user']
+                        if comment['content'] != "":
+                            comment_data_filtered.append(comment)
+
+
+                    json_obj[i]['comments'] = comment_data_filtered
+                    
+                    json_obj[i]['description'] = self.parse_desc(json_obj[i]['description'])
+
+                    if('id' in json_obj[i] and not isinstance(json_obj[i]['id'], str)):
+                        json_obj[i]['description'] = json_obj[i]['description'] + "\n\nSearchlight Portal Link: https://portal-digitalshadows.com/triage/alert-incidents/" + str(json_obj[i]['id'])
+                
+                    self.AS_obj.post_data(json.dumps(json_obj[i]), constant.LOG_NAME)
+
+    def get_data(self):
+        """
+            getting the incident and alert data from digital shadows
+        """
+        triage_id = []
+
+        try:
+            if(isinstance(self.event, int)):
+                event_data = self.DS_obj.get_triage_events_by_num(self.event)
+                #calculating the max event number from current batch to  use in next call
+                if(event_data):
+                    max_event_num = max([e['event-num'] for e in event_data])
+
+            else:
+                event_data = self.DS_obj.get_triage_events(self.before_time, self.after_time)
+                #calculating the max event number from current batch to  use in next call
+                if(event_data):
+                    max_event_num = max([e['event-num'] for e in event_data])
+                    logger.info("First poll from event number " + str(event_data[0]['event-num']))
+                    logger.info("Total number of events are " + str(len(event_data)))
+                else:
+                    logger.info("There are no new events")
+                    return [], int
+
+            
+            for event in event_data:
+                if(event is not None and event['triage-item-id'] not in triage_id):
+                    triage_id.append(event['triage-item-id'])
+
+            logger.info(triage_id)
+        except Exception:            
```

Comment:
I don't think we should be catching this at all. We should allow exceptions to propagate up to `poll`. As it stands we'll continue and attempt to `get_triage_items` with an empty list of triage IDs if we catch here.